pkg/rpc/response/errors.go

```diff
@@ -27,7 +27,9 @@ var (
 	// ErrUnableToVerify represents SubmitError with code -503
 	ErrUnableToVerify = NewSubmitError(-503, "The block cannot be validated.")
 	// ErrValidationFailed represents SubmitError with code -504
-	ErrValidationFailed = NewSubmitError(-504, "Block or transaction validation failed.")
+	ErrValidationFailed = func(cause error) *Error {
```

Comment:
I like the intentions of this PR, but we can do better with implementation. It should be possible to do `errors.Is(..., response.ErrValidationFailed)` client-side.

Reply:
Fixed.